runtime-sdk/src/modules/accounts/mod.rs

```diff
@@ -952,6 +952,32 @@ impl module::TransactionHandler for Module {
         Ok(())
     }
 
+    fn after_handle_call<C: TxContext>(ctx: &mut C) -> Result<(), modules::core::Error> {
```

Comment:
Can this result in the move from tx fee accumulator to block fee accumulator not happening in case the transaction fails before call dispatch (but fees have been charged as they are charged in `authenticate_tx`)? For example, in case the (encrypted) call cannot be decoded or the `before_handle_call` validation fails.

We should also add a test for this case.

Reply:
I guess the problem is also about emitting events as they will be emitted in batch context instead of tx? Maybe we could add another hook (or call `after_handle_call` with an additional result argument) which would be called from `dispatch_tx_opts` in the failed call handler?

Reply:
Oh right, I also wanted to emit the fee-events as transaction tags not as block tags. But I missed that `after_handle_call` is not always called.

So likely I would need a new hook that is called with transaction context, regardless of any errors during the tx processing, then I can `emit_unconditonally` events there.

Reply:
I didn't see your reply before posting my last two replies :grin: I think updating the `after_handle_call` should be best?

Reply:
Yeah probably should be. Maybe with an additional argument signalling the status (success vs. failure).

Reply:
Updated, and also added a test (see 2nd commit).